sematic/api/endpoints/external_resources.py

```diff
@@ -79,7 +82,9 @@ def activate_resource_endpoint(
             "No such resource: {}".format(resource_id), HTTPStatus.NOT_FOUND
         )
     try:
-        activated = record.resource.activate(is_local=False)
+        resource: AbstractExternalResource = record.resource
+        activated = resource.activate(is_local=False)
+        resource.track_usage(user=user, event="activated")
```

Comment:
I would use something other than mixins; or it might make it harder to trace code and reduce readability. (developer might wonder why `resource` has a method of `track_usage`, where was it defined? Afterall, The `resource` in the context is a `AbstractExternalResource` , not `RayCluster`, it shouldn't have `track_usage`) 

How about doing an API like this:
```
posthog_track_usage(resource, user=user, event="activated)
```